src/Component/Context/Context.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Resource\Context;
+
+/**
+ * @implements \IteratorAggregate<object>
+ */
+final class Context implements \IteratorAggregate
+{
+    /** @var array<class-string, object> */
+    private array $optionMap;
+
+    public function __construct(object|array ...$options)
+    {
+        $map = [];
+        foreach ($options as $option) {
+            if (\is_array($option)) {
```

Comment:
@mtarld I've improved that part to pass an interface as a key instead of the class.
WDYT?

Reply:
Just wondering, why do you need to have an interface as a key there?
Be careful btw as it might introduce collisions

Reply:
Not needed now, as discussed together.

Reply:
Encapsulate the service in a context option seems to be perfect.

```
final class InputOption
{
    public function __construct(private InputInterface $input)
    {
    }

    public function input(): InputInterface
    {
        return $this->input;
    }
}
```

Reply:
I'm just wondering why won't we prefer to have

```php
$inputOption = $context->get(InputOption::class);
$outputOption = $context->get(OutputOption::class);

$ui = new SymfonyStyle($inputOption->input(), $outputOption->output());
```

with still being sure, that these classes exists :)

Reply:
Yes but not be sure it is in the context. Cause it depends to the... Context (cli or http for exemple)